src/widgets/conditions/condition-record.component.tsx

```diff
@@ -1,39 +1,77 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { useRouteMatch } from "react-router-dom";
 import { getConditionByUuid } from "./conditions.resource";
 import { createErrorHandler } from "@openmrs/esm-error-handling";
+import { useCurrentPatient, newWorkspaceItem } from "@openmrs/esm-api";
 import dayjs from "dayjs";
 import styles from "./condition-record.css";
 import SummaryCard from "../../ui-components/cards/summary-card.component";
-import { useCurrentPatient } from "@openmrs/esm-api";
+import { ConditionsForm } from "./conditions-form.component";
 
 export default function ConditionRecord(props: ConditionRecordProps) {
-  const [patientCondition, setPatientCondition] = React.useState(null);
+  const [patientCondition, setPatientCondition] = useState(null);
   const [isLoadingPatient, patient, patientUuid] = useCurrentPatient();
   const match = useRouteMatch();
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (!isLoadingPatient && patient) {
       const sub = getConditionByUuid(match.params["conditionUuid"]).subscribe(
-        condition => setPatientCondition(condition),
+        ({ resource }) => setPatientCondition(resource),
         createErrorHandler()
       );
       return () => sub.unsubscribe();
     }
   }, [isLoadingPatient, patient, match.params]);
 
+  const openEditConditionsWorkspaceTab = (
+    componentName,
+    title,
+    conditionUuid,
+    name,
+    clinicalStatus,
+    onsetDateTime
+  ) => {
+    newWorkspaceItem({
+      component: componentName,
+      name: title,
+      props: {
+        match: {
+          params: {
+            conditionUuid,
+            name,
+            clinicalStatus,
+            onsetDateTime
+          }
+        }
+      },
+      inProgress: false,
+      validations: (workspaceTabs: any[]) =>
+        workspaceTabs.findIndex(tab => tab.component === componentName)
+    });
+  };
+
   function displayCondition() {
     return (
       <>
         <SummaryCard
           name="Condition"
           styles={{ width: "100%" }}
-          editBtnUrl={`/patient/${patientUuid}/chart/conditions/edit`}
+          editComponent={ConditionsForm}
+          showComponent={() => {
+            openEditConditionsWorkspaceTab(
+              ConditionsForm,
+              "Edit Conditions",
+              patientCondition.id,
+              patientCondition.code.text,
+              patientCondition.clinicalStatus,
+              patientCondition.onsetDateTime
+            );
+          }}
         >
           <div className={`omrs-type-body-regular ${styles.conditionCard}`}>
             <div>
               <p className="omrs-type-title-3" data-testid="condition-name">
-                {patientCondition.resource.code.text}
+                {patientCondition.code.text}
```

Comment:
I am surprised here. Below we use `patientCondition?.` implying that `patientCondition` may be undefined - now we access `code.text` here without the conditional member operator. Is this desired?

Should it be `patientCondition?.code.text` instead?

Reply:
Sure. This was an oversight on my part. Corrected.